sdk/core/azure-core/azure/core/exceptions.py

```diff
@@ -65,7 +65,7 @@ def raise_with_traceback(exception, *args, **kwargs):
     message = kwargs.pop("message", "")
     exc_type, exc_value, exc_traceback = sys.exc_info()
     # If not called inside a "except", exc_type will be None. Assume it will not happen
-    exc_msg = "{}, {}: {}".format(message, exc_type.__name__, exc_value)  # type: ignore
+    exc_msg = "{}, {}: {}".format(message, getattr(exc_type, '__name__'), exc_value)
```

Comment:
See the comment line 67. To make it better, we might just check if it's None and then raise, but using `getattr` is actually hiding the problem (it's another form of ignore :)), since Mypy can't check `getattr` by design (dynamic)